crypto/hash/htr/hashtree.go

```diff
@@ -0,0 +1,17 @@
+package htr
+
+import (
+	"github.com/prysmaticlabs/gohashtree"
+)
+
+// VectorizedSha256 takes a list of roots and hashes them using CPU
+// specific vector instructions. Depending on host machine's specific
+// hardware configuration, using this routine can lead to a significant
+// performance improvement compared to the default method of hashing
+// lists.
+func VectorizedSha256(inputList [][32]byte, outputList [][32]byte) {
```

Comment:
I think instead of having this function here, I should export a function in gohashtree that calls directly the assembly. Here we are adding an extra hop without any need. 

Also, panicking on a hashing error would make us crash. Can we not panic here?

Reply:
We need to throw an error here, the other alternative is we change the function signature to return an error, which will necessitate all upstream functions also having an error. The only 2 error modes from which it can crash are here:
https://github.com/prysmaticlabs/gohashtree/blob/main/hash.go#L32

Would it be fine to expect the caller to handle this ?